pollard_test.go

```diff
@@ -1289,3 +1290,76 @@ func compareNodeMap(mapA, mapB map[miniHash]*polNode) error {
 
 	return fmt.Errorf(str)
 }
+
+// Test that the cached nodes are correct.
+func TestProofCacheNodes(t *testing.T) {
+
+	// Create a new accumulator with 10 leaves.
+	adds := make([]Leaf, 10)
+	for i := range adds {
+		adds[i] = Leaf{Hash: sha256.Sum256([]byte{uint8(i)})}
+	}
+
+	// Create the accumulator and add one leaf at a time.
+	p := NewAccumulator(true)
```

Comment:
Passing `true` to `NewAccumulator` will make it so that nothing's being tested. Everything is being cached anyways.